src/evo/mnauth.cpp

```diff
@@ -34,7 +34,15 @@ void CMNAuth::PushMNAUTH(CNode* pnode, CConnman& connman)
         // It does not protect against:
         //   node1 -> Eve -> node2
         // This is ok as we only use MNAUTH as a DoS protection and not for sensitive stuff
-        signHash = ::SerializeHash(std::make_tuple(*activeMasternodeInfo.blsPubKeyOperator, pnode->receivedMNAuthChallenge, pnode->fInbound));
+        int nOurNodeVersion{PROTOCOL_VERSION};
+        if (Params().NetworkIDString() != CBaseChainParams::MAIN && gArgs.IsArgSet("-pushversion")) {
```

Comment:
I presume we don't intend to keep this, so maybe add a TODO to remove in either rc4 or final 0.16

Reply:
Nope, this is needed because we imitate old nodes in `feature_llmq_simplepose.py` and they should send `mnauth` in correct format cause otherwise other nodes won't let them connect. Same for `ProcessMessage` - if we pretend to be an older node we should be expecting other nodes to send us `mnauth` in old format.